### PR TITLE
⚡ Add option to disable controls on video and audio

### LINKED ISSUE #1485: Option to disable controls on video and audio bubbles

### PR DESCRIPTION
**Pull request by michaldziuba03**
This PR adds option to remove controls for Videos (for now only URL type) and Audio bubbles.

https://github.com/baptisteArno/typebot.io/assets/43048524/6b5282f2-bd55-4e8a-ad26-34cd2db77835

> Enable sound in video preview

Closes #1485

### CODE REVIEW

File: packages/embeds/js/src/features/blocks/bubbles/video/components/VideoBubble.tsx [hunk 1]
```diff
@@ -77,9 +77,12 @@ export const VideoBubble = (props: Props) => {
               }
             >
               <video
-                autoplay={props.onTransitionEnd ? false : true}
+                autoplay={
+                  (props.onTransitionEnd ? false : true) ||
+                  props.content?.hideControls
+                }
```
Comment (michaldziuba03) [inline packages/embeds/js/src/features/blocks/bubbles/video/components/VideoBubble.tsx:83]: I decided to autoplay the video as it would be impossible to play the video without controls.

**Comment by michaldziuba03** [comment]
@baptisteArno ready for review

File: packages/embeds/js/src/features/blocks/bubbles/video/components/VideoBubble.tsx (hunk 1 above)
Comment (baptisteArno) [reply packages/embeds/js/src/features/blocks/bubbles/video/components/VideoBubble.tsx:83]: This check is here to avoid replaying all the media bubbles when we return to a chat session (after a refresh or else).

So this is need to stay as is otherwise, all the media without controls would play all at once on a page refresh if the chat state is saved

File: packages/schemas/features/blocks/bubbles/audio/schema.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { BubbleBlockType } from '../constants'
 export const audioBubbleContentSchema = z.object({
   url: z.string().optional(),
   isAutoplayEnabled: z.boolean().optional(),
+  hideControls: z.boolean().default(false).optional(),
```
Comment (baptisteArno) [inline packages/schemas/features/blocks/bubbles/audio/schema.ts:8]: I'd prefer a field named `areControlsDisplayed` to stay consistent with the repo boolean var convention

File: apps/builder/src/i18n/en.json [hunk 3]
```diff
@@ -465,6 +466,7 @@
   "video.aspectRatioInput.moreInfoTooltip": "Example: \"16/9\" or \"9/16\"",
   "video.maxWidthInput.label": "Max width",
   "video.maxWidthInput.moreInfoTooltip": "Example: \"300px\" or \"100%\"",
+  "video.hideControlsInput.label": "Hide video controls",
```
Comment (baptisteArno) [inline apps/builder/src/i18n/en.json:469]: Remove i18n manual file editting, it will be done on another PR :)

File: apps/builder/src/features/blocks/bubbles/audio/components/AudioBubbleForm.tsx [hunk 4]
```diff
@@ -82,6 +86,11 @@ export const AudioBubbleForm = ({
           }
           onCheckChange={updateAutoPlay}
         />
+        <SwitchWithLabel
+          label={t('editor.blocks.bubbles.audio.settings.hideControls.label')}
```
Comment (baptisteArno) [inline apps/builder/src/features/blocks/bubbles/audio/components/AudioBubbleForm.tsx:90]: Don't use i18n for now

**Review by baptisteArno — CHANGES_REQUESTED** [review]
Thank you for that work 👌

I'd prefer if the option is an enabled instead of a disabler. So instead of having a `Hide controls` option, I'd prefer having a `Display controls` option that is `true` by default.

You can add a default option in the `constants` file (i.e. https://github.com/baptisteArno/typebot.io/blob/35300eaf340c9bcae41f7b004c682bb8939684ff/packages/schemas/features/blocks/bubbles/audio/constants.ts#L1)

File: packages/embeds/js/src/features/blocks/bubbles/video/components/VideoBubble.tsx [hunk 5]
```diff
@@ -77,9 +77,19 @@ export const VideoBubble = (props: Props) => {
...
               <video
-                autoplay={props.onTransitionEnd ? false : true}
+                autoplay={
+                  props.onTransitionEnd
+                    ? !(
+                        props.content?.areControlsDisplayed ??
+                        defaultVideoBubbleContent.areControlsDisplayed
+                      )
+                    : false
+                }
```
Comment (michaldziuba03) [inline packages/embeds/js/src/features/blocks/bubbles/video/components/VideoBubble.tsx:87]: I am not sure if this is correct

I took logic from `AudioBubble.tsx`

https://github.com/baptisteArno/typebot.io/blob/e755f08b403cdb4193ce93b1bb02098444f99a89/packages/embeds/js/src/features/blocks/bubbles/audio/components/AudioBubble.tsx#L61-L66

**Comment by baptisteArno** [comment]
I guess disabling the controls on the audio element does not make sense as it hides the audio element altogether!

**Comment by baptisteArno** [comment]
Thanks again for the work on this. I am closing this in favor of #1631 🙏
